content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,486 @@
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+
+	pageReviewURL := fmt.Sprintf("%s/review/%s?uri=%s/data.json&recursive=false", environment, collectionID, cdIDURI)
+	req, err := http.NewRequestWithContext(ctx, "POST", pageReviewURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+		return errMessage
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(FlorenceToken, ctx.Value(ZebedeeToken).(string))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	if resp.StatusCode > 400 {
+		errMessage := fmt.Errorf("failed to approve CDID. Error In API: %v. Status Code: %s", err, resp.Status)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	defer resp.Body.Close()
+	return nil
+}
+
+// POST http://localhost:8081/zebedee/content
+//		/test-4c153d5de19c33be41b772b9d6c27dfc917bda71f2d8925e6b95e17da3a7d8cf
+//		?uri=inflationandpriceindices/timeseries/mb55/mm22/data.json
+//		&overwriteExisting=true
+
+func addCDIDToCollection(ctx context.Context, client *http.Client, environment string, collectionID string, cdIDLocation string, data string) error {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+
+	pageURL := fmt.Sprintf("%s/zebedee/content/%s?uri=%s/data.json&overwriteExisting=true", environment, collectionID, cdIDURI)
+	req, err := http.NewRequestWithContext(ctx, "POST", pageURL, bytes.NewBuffer([]byte(data)))
+	if err != nil {
+		errMessage := fmt.Errorf("failed to page adding to collection request. Error: %v", err)
+		log.Error(errMessage)
+		return errMessage
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(ZebedeeToken, ctx.Value(ZebedeeToken).(string))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to add collection to page. Error: %v", err)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	if resp.StatusCode > 400 {
+		errMessage := fmt.Errorf("failed to add collection to page. Error In API: %v. Status Code: %s", err, resp.Status)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	return nil
+}
+
+func fetchDataForCDID(ctx context.Context, client *http.Client, environment string, location string) (string, error) {
+
+	cdIDDataURL := fmt.Sprintf("%s/data", location)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", cdIDDataURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to prepare fetch CDID request. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		if err != nil {
+			errMessage := fmt.Errorf("failed to fetch CDID data from API. Error: %v", err)
+			log.Error(errMessage)
+
+			return "", errMessage
+		}
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Before you read it you should check the response status is 200 - anything else is an error.